mozci/data/sources/treeherder/__init__.py

```diff
@@ -97,3 +102,32 @@ def run_pushes(self, branch, nb=15) -> List[Dict]:
             }
             for p in response["results"]
         ]
+
+    def get_job_from_task(self, task: Task) -> dict:
+        """Map a task to its Treeherder job, based on the Job list API.
+        The endpoint returns a list of jobs, but only 1 result should be present using the `task_id` parameter.
+        """
+        data = self._run_query(f"/jobs/?task_id={task.id}")
+        job = next(iter(data.get("results", [])), None)
+        if not job:
+            raise JobUnavailable
+        if "job_property_names" not in data:
+            raise Exception(
+                "`job_property_names` is missing from /jobs/ treeherder API response"
+            )
+        return {key: value for key, value in zip(data["job_property_names"], job)}
+
+    def check_logs_parsed(self, job_id: int, branch: str = "autoland") -> bool:
+        """Check if the logs of a specific job have been parsed already.
+        Raises JobUnavailable in case no job log are found or some are still in a pending state.
+        """
+        data = self._run_query(f"/project/{branch}/job-log-url/?job_id={job_id}")
+        if not data:
+            raise JobUnavailable
+        if any(job_log_url.get("parse_status") == "pending" for job_log_url in data):
+            raise JobUnavailable
+        return all(job_log_url.get("parse_status") == "parsed" for job_log_url in data)
```

Comment:
The return value is not used - is this intended?

Reply:
I made the changes but still have a question. As this check only concern the main failure, could such states affect the `bug_suggestion` response from Treeherder ?

Reply:
There cannot be bug suggestions for `failed` and `skipped_size`, behavior should be unchanged.